In [2]:
# INSTALL PACKAGES ####
install.packages("tidyverse")
install.packages("gmodels")

# USE PACKAGES ####
library("tidyverse")
library("gmodels")


The downloaded binary packages are in
	/var/folders/wk/6why77bn1kn0l0pkd4vd3zl00000gn/T//Rtmpfl2mTb/downloaded_packages


In [4]:
# READ IN DATA
loans <- read.csv('../data/loans.csv')

In [6]:
# VIEW DATA
head(loans)

X term       loan_status Date      home_ownership
1 1  36 months Fully Paid  12/1/2011 RENT          
2 2  60 months Charged Off 12/1/2011 RENT          
3 3  36 months Fully Paid  12/1/2011 RENT          
4 4  36 months Fully Paid  12/1/2011 RENT          
5 5  60 months Current     12/1/2011 RENT          
6 6  36 months Fully Paid  12/1/2011 RENT

## Part I
Please answer the following question.

Does the term of the loan influence loan status? If so, how?

In [13]:
CrossTable(loans$term, loans$loan_status, fisher=FALSE, chisq = TRUE, expected = TRUE, sresid=TRUE, format="SPSS")

# 36 month term loans are more likely to be fully paid off and less likely to be charged off or current.  
# The reverse is true for 60 month loans.



   Cell Contents
|-------------------------|
|                   Count |
|         Expected Values |
| Chi-square contribution |
|             Row Percent |
|          Column Percent |
|           Total Percent |
|            Std Residual |
|-------------------------|

Total Observations in Table:  21957 

             | loans$loan_status 
  loans$term | Charged Off  |     Current  |  Fully Paid  |   Row Total | 
-------------|-------------|-------------|-------------|-------------|
   36 months |       2029  |          0  |      14964  |      16993  | 
             |   2540.011  |    388.509  |  14064.480  |             | 
             |    102.808  |    388.509  |     57.530  |             | 
             |     11.940% |      0.000% |     88.060% |     77.392% | 
             |     61.822% |      0.000% |     82.342% |             | 
             |      9.241% |      0.000% |     68.151% |             | 
             |    -10.139  |    -19.711  |      7.585  |             | 
-------

## Part II
Please answer the following question.

How has the ability to own a home changed after 2009?

To reformat the Date column into a new column, use this code:

```
loans$DateR <- as.Date(paste(loans$Date), "%m/%d/%Y")
```

In [9]:
loans$DateR <- as.Date(paste(loans$Date), "%m/%d/%Y")

In [11]:
head(loans)

X term       loan_status Date      home_ownership DateR     
1 1  36 months Fully Paid  12/1/2011 RENT           2011-12-01
2 2  60 months Charged Off 12/1/2011 RENT           2011-12-01
3 3  36 months Fully Paid  12/1/2011 RENT           2011-12-01
4 4  36 months Fully Paid  12/1/2011 RENT           2011-12-01
5 5  60 months Current     12/1/2011 RENT           2011-12-01
6 6  36 months Fully Paid  12/1/2011 RENT           2011-12-01

In [14]:
loans$Issue_dateR <- as.Date(loans$DateR, format="%b-%d")

In [15]:
loans1 <- separate(loans, Issue_dateR, c("Ignore", "Issue_Month", "Issue_Year"), sep="-")

In [16]:
loans1$Issue_MonthR <- NA
loans1$Issue_MonthR[loans1$Issue_Year == "07"] <- 0
loans1$Issue_MonthR[loans1$Issue_Year == "11"] <- 1

In [17]:
loans1$RentvOwn <- NA
loans1$RentvOwn[loans1$home_ownership == "RENT"] <- 0
loans1$RentvOwn[loans1$home_ownership == "OWN"] <- 1

In [21]:
loans1

X     term       loan_status Date      home_ownership DateR      Ignore
1      1     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
2      2     60 months Charged Off 12/1/2011 RENT           2011-12-01 2011  
3      3     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
4      4     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
5      5     60 months Current     12/1/2011 RENT           2011-12-01 2011  
6      6     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
7      7     60 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
8      8     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
9      9     60 months Charged Off 12/1/2011 OWN            2011-12-01 2011  
10    10     60 months Charged Off 12/1/2011 RENT           2011-12-01 2011  
11    11     60 months Fully Paid  12/1/2011 OWN            2011-12-01 2011  
12    12     36 months Fully Paid  12/1/2011 OWN            2011-12-01 2011  
13    13     36 months Charged Off 12/1/2011 RENT           2011-12-01 2011  
14    14     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
15    15     36 months Charged Off 12/1/2011 RENT           2011-12-01 2011  
16    16     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
17    17     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
18    18     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
19    19     60 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
20    20     36 months Charged Off 12/1/2011 RENT           2011-12-01 2011  
21    21     36 months Fully Paid  12/1/2011 OWN            2011-12-01 2011  
22    22     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
23    23     36 months Charged Off 12/1/2011 RENT           2011-12-01 2011  
24    24     36 months Charged Off 12/1/2011 RENT           2011-12-01 2011  
25    25     60 months Charged Off 12/1/2011 RENT           2011-12-01 2011  
26    26     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
27    27     60 months Current     12/1/2011 RENT           2011-12-01 2011  
28    28     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
29    29     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
30    30     36 months Fully Paid  12/1/2011 RENT           2011-12-01 2011  
⋮     ⋮     ⋮          ⋮           ⋮         ⋮              ⋮          ⋮     
21928 21928  36 months Fully Paid  8/1/2007  OWN            2007-08-01 2007  
21929 21929  36 months Fully Paid  8/1/2007  OWN            2007-08-01 2007  
21930 21930  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21931 21931  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21932 21932  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21933 21933  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21934 21934  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21935 21935  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21936 21936  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21937 21937  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21938 21938  36 months Fully Paid  7/1/2007  RENT           2007-07-01 2007  
21939 21939  36 months Fully Paid  7/1/2007  RENT           2007-07-01 2007  
21940 21940  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21941 21941  36 months Fully Paid  8/1/2007  OWN            2007-08-01 2007  
21942 21942  36 months Fully Paid  8/1/2007  RENT           2007-08-01 2007  
21943 21943  36 months Fully Paid  8/1/2007  OWN            2007-08-01 2007  
21944 21944  36 months Fully Paid  7/1/2007  RENT           2007-07-01 2007  
21945 21945  36 months Fully Paid  7/1/2007  RENT           2007-07-01 2007  
21946 21946  36 months Fully Paid  7/1/2007  RENT           2007-07-01 2007  
21947 21947  36 months Fully

In [20]:
CrossTable(loans1$RentvOwn, loans1$Issue_MonthR, fisher=TRUE, chisq = TRUE, mcnemar = TRUE, expected = TRUE, sresid=TRUE, format="SPSS")

# Although the overall test is significant, when standardized residuals are examined, 
# it looks like the number of people renting to owning houses is similar from 2007 to 2011.


Warning message in chisq.test(t, correct = FALSE, ...):
“Chi-squared approximation may be incorrect”



   Cell Contents
|-------------------------|
|                   Count |
|         Expected Values |
| Chi-square contribution |
|             Row Percent |
|          Column Percent |
|           Total Percent |
|            Std Residual |
|-------------------------|

Total Observations in Table:  21957 

                | loans1$DateR 
loans1$RentvOwn | 2007-06-01  | 2007-07-01  | 2007-08-01  | 2007-09-01  | 2007-10-01  | 2007-11-01  | 2007-12-01  | 2008-01-01  | 2008-02-01  | 2008-03-01  | 2008-04-01  | 2008-05-01  | 2008-06-01  | 2008-07-01  | 2008-08-01  | 2008-09-01  | 2008-10-01  | 2008-11-01  | 2008-12-01  | 2009-01-01  | 2009-02-01  | 2009-03-01  | 2009-04-01  | 2009-05-01  | 2009-06-01  | 2009-07-01  | 2009-08-01  | 2009-09-01  | 2009-10-01  | 2009-11-01  | 2009-12-01  | 2010-01-01  | 2010-02-01  | 2010-03-01  | 2010-04-01  | 2010-05-01  | 2010-06-01  | 2010-07-01  | 2010-08-01  | 2010-09-01  | 2010-10-01  | 2010-11-01  | 2010-12-01  | 2011-01-01  | 2011-02-01  | 2011-03-01 

ERROR: Error in mcnemar.test(t, correct = FALSE): 'x' must be square with at least two rows and columns


## Part III
Please answer the following question.

The news just ran a story that in America, only 15% of homes are fully paid, 10% have given up on paying it back ("charged off"), and 75% are current in the loan. Does it seem likely that the data for this hands-on came from the larger population of America?